### PR TITLE
[Satel] Migration to ESH serial API

### PR DESCRIPTION
**Pull request by druciak**
Signed-off-by: Krzysztof Goworek <krzysztof.goworek@gmail.com>

This PR removes dependency to gnu.io and migrates the code to use ESH serial API.

### CODE REVIEW

File: bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/protocol/IntRSModule.java [hunk 1]
```diff
@@ -17,19 +17,18 @@
 import java.io.OutputStream;
 import java.util.TooManyListenersException;
 
-import javax.naming.ConfigurationException;
-
+import javax.validation.constraints.NotNull;
```
Comment (martinvw) [inline bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/protocol/IntRSModule.java]: This is the wrong annotation, we use NonNull and its from another package :-)

File: bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/protocol/IntRSModule.java [hunk 2]
```diff
@@ -40,32 +39,37 @@
 
     private final Logger logger = LoggerFactory.getLogger(IntRSModule.class);
 
-    private String port;
+    private final @NotNull String port;
```
Comment (martinvw) [inline bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/protocol/IntRSModule.java]: Please do not use NonNull but use NonNullByDefault as [described in the coding guidelines](https://www.openhab.org/docs/developer/development/guidelines.html#a-code-style)

File: bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/protocol/SatelModule.java [hunk 3]
```diff
@@ -88,7 +88,11 @@
      */
     protected static class ConnectionFailureException extends Exception {
 
-        private static final long serialVersionUID = 1L;
+        private static final long serialVersionUID = 2L;
```
Comment (martinvw) [inline bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/protocol/SatelModule.java]: That is indeed how its supposed to be used (but most people tent to forget to change it )

File: bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/protocol/IntRSModule.java (hunk 1 above)
Comment (druciak) [reply bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/protocol/IntRSModule.java]: My mistake, done.

File: bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/protocol/IntRSModule.java (hunk 2 above)
Comment (druciak) [reply bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/protocol/IntRSModule.java]: Done.

File: bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/protocol/IntRSModule.java [hunk 4]
```diff
@@ -17,55 +17,60 @@
...
  *
  * @author Krzysztof Goworek - Initial contribution
  */
+@NonNullByDefault
 public class IntRSModule extends SatelModule {
 
     private final Logger logger = LoggerFactory.getLogger(IntRSModule.class);
 
-    private String port;
+    private final String port;
```
Comment (martinvw) [inline bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/protocol/IntRSModule.java]: 👍 I like those final fields

**Comment by druciak** [comment]
Thanks @martinvw for very quick merge!

**Pull request merged**